Fix I2CPullups API mismatch causing 3 test failures

## Summary

3 tests in `test_schematic_blocks.py` are failing due to API mismatch between the tests and the `I2CPullups` implementation.

## Failing Tests

```
FAILED tests/test_schematic_blocks.py::TestI2CPullupsMocked::test_i2c_pullups_with_filter_caps
FAILED tests/test_schematic_blocks.py::TestI2CPullupsFactoryFunctions::test_create_i2c_pullups_default
FAILED tests/test_schematic_blocks.py::TestI2CPullupsFactoryFunctions::test_create_i2c_pullups_custom_value
```

## Error Details

1. **test_i2c_pullups_with_filter_caps**:
   ```
   TypeError: I2CPullups.__init__() got an unexpected keyword argument 'filter_cap'
   ```

2. **test_create_i2c_pullups_default** and **test_create_i2c_pullups_custom_value**:
   ```
   TypeError: create_i2c_pullups() got an unexpected keyword argument 'ref'
   ```

## Analysis

The tests expect:
- `I2CPullups(filter_cap="100pF")` - filter capacitor parameter
- `create_i2c_pullups(ref="R1")` - reference designator parameter

But the current implementation doesn't support these parameters.

## Resolution Options

1. **Add missing parameters to implementation** - If these features are desired
2. **Remove/update tests** - If the tests are incorrect

## Acceptance Criteria

- [ ] All 3 failing tests pass
- [ ] API is consistent between implementation and tests

## Part of

#155 v0.5.0 Release Tracking

--- a/tests/test_schematic_blocks.py
+++ b/tests/test_schematic_blocks.py
@@ -2288,8 +2288,8 @@ class TestI2CPullupsMocked:
 
     def test_i2c_pullups_with_filter_caps(self, mock_schematic):
         """Create I2C pull-ups with filter capacitors."""
-        pullups = I2CPullups(mock_schematic, x=100, y=100, filter_cap="100pF")
-        assert pullups.filter_cap == "100pF"
+        pullups = I2CPullups(mock_schematic, x=100, y=100, filter_caps="100pF")
+        assert pullups.filter_caps_value == "100pF"
 
     def test_i2c_pullups_ports(self, mock_schematic):
         """Verify I2C pull-ups port definitions."""
@@ -2314,12 +2314,14 @@ class TestI2CPullupsFactoryFunctions:
 
     def test_create_i2c_pullups_default(self, mock_schematic):
         """Create I2C pull-ups with default values."""
-        pullups = create_i2c_pullups(mock_schematic, x=100, y=100, ref="R1")
+        pullups = create_i2c_pullups(mock_schematic, x=100, y=100)
         assert isinstance(pullups, I2CPullups)
 
     def test_create_i2c_pullups_custom_value(self, mock_schematic):
-        """Create I2C pull-ups with custom resistor value."""
-        pullups = create_i2c_pullups(mock_schematic, x=100, y=100, ref="R1", resistor_value="10k")
+        """Create I2C pull-ups with custom resistor value using class directly."""
+        # Use I2CPullups directly for custom resistor values
+        # Factory function uses speed presets, not custom values
+        pullups = I2CPullups(mock_schematic, x=100, y=100, resistor_value="10k")
         assert pullups.resistor_value == "10k"
 
 

